apps/web/src/server/app/lpa/lpa.controller.js

```diff
@@ -1,6 +1,33 @@
 import { to } from 'planning-inspectorate-libs';
-import { findAllIncomingIncompleteQuestionnaires } from './lpa.service.js';
+import { findAllIncomingIncompleteQuestionnaires, findQuestionnaireById } from './lpa.service.js';
 import { lpaRoutesConfig as routes } from '../../config/routes.js';
+import { camelCase, upperFirst } from 'lodash-es';
+import { checkboxDataToCheckValuesObject, appealSiteObjectToText, makeDownloadLinksDataForTemplate } from '../../lib/helpers.js';
+
+/**
+ * Create an array of row data for consumption in nunjucks template govukTable component, using the supplied data array from the LPA service
+ *
+ * @param {Array<object>} questionnairesList - array of questionnaire data items returned from findAllIncomingIncompleteQuestionnaires service method
+ * @returns {Array<object>} - array of row data objects for consumption in nunjucks template
+ */
+function makeQuestionnairesListRowsForTemplate (questionnairesList) {
+	const rows = [];
+
+	// eslint-disable-next-line unicorn/no-array-for-each
+	questionnairesList.forEach((item) => {
```

Comment:
My personal opinion is formatting and html manipulation shouldn't be in the controller, as it becomes more like a ViewModel and and I don't think that's the design we're going for. I think controllers should just be handling the incoming request and rendering a template with data, then nunjucks does any template manipulation.

I would personally be passing the questionnaires list to the template, and then using a nunjucks macro or filter. Perhaps @radum has an opinion.

Reply:
I agree, although I didn't had strong feelings when we started due to the time we have, I think its a good thing to do from the start. Using macros should handle these situations, and if the data is way to skewed we can talk to the API devs.

Reply:
Thanks guys, good to have clarity on the approach. I've refactored this to use a macro and it does seem cleaner. Is there a rule of thumb for when to manipulate data in the service/controller vs. in the template? For example, should any and all manipulation of data occur only in the templates, or just when it involves rendering strings/HTML?

I assume we will use this approach across the whole codebase. If so there are a few bits we should go back and refactor; I'll mark these with TODO comments.

Reply:
After chatting with @radum I've refactored code in this PR which was generating html or otherwise performing template-specific formatting of the data. This involved adding a couple of macros and nunjucks helpers, but the approach does feel cleaner overall. I marked some other bits that were out of scope for this PR with TODO comments.